src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -225,6 +245,22 @@ subroutine init_atm_static(mesh, dims, configs)
  areaTriangle = areaTriangle * sphere_radius**2.0
  kiteAreasOnVertex = kiteAreasOnVertex * sphere_radius**2.0
 
+!
+! Initialize the KD-Tree
+!
+allocate(kd_points(nCells))
+do i = 1, nCells
+    allocate(kd_points(i) % point(3))
+    kd_points(i) % point = (/xCell(i), yCell(i), zCell(i)/)
+    kd_points(i) % id = i ! Cell ID
+enddo
+tree => null()
+tree => mpas_kd_construct(kd_points, 3)
+if (.not. associated(tree)) then
+    call mpas_log_write('Error creating the KD-Tree', messageType=MPAS_LOG_CRIT)
+    return
```

Comment:
Can we indent this code block by one column to match the surrounding code?

I think the `return` statement is also unnecessary, since a critical error message will stop execution of the program.